File: Sliding Window/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0 
        buy_price = 10**4+1
        
        for price in prices:
            buy_price = min(price, buy_price)
            res = max(res, price-buy_price)
            
        return res

File: Sliding Window/121. Best Time to Buy and Sell Stock.c
#define MAX(x, y) (((x) > (y)) ? (x) : (y))
#define MIN(x, y) (((x) < (y)) ? (x) : (y))

int maxProfit(int* prices, int pricesSize){
    int res = 0;
    int buy_price = 1000000;
    for (int i = 0; i < pricesSize; i++){
        buy_price = MIN(buy_price, prices[i]);
        res = MAX(res, prices[i] - buy_price);
    }
    return res;
}
